client/pluginmanager/csimanager/manager.go

```diff
@@ -0,0 +1,153 @@
+package csimanager
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/dynamicplugins"
+	"github.com/hashicorp/nomad/client/pluginmanager"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// defaultPluginResyncPeriod is the time interval used to do a full resync
+// against the dynamicplugins, to account for missed updates.
+const defaultPluginResyncPeriod = 30 * time.Second
```

Comment:
It looks like the run loop in this file is the only place we create a new instance manager. So when `csiPluginSupervisorHook` registers the plugin, we don't get the instance manager for as long as 30s?

Reply:
eventually we'll be subscribing to updates from the plugin manager too - just didn't want to put more machinery in this PR

Reply:
Sounds good.